airbyte-integrations/connectors/source-amplitude/source_amplitude/streams.py

```diff
@@ -206,3 +173,29 @@ def next_page_token(self, response: requests.Response) -> Optional[Mapping[str,
 
     def path(self, **kwargs) -> str:
         return f"{self.api_version}/export"
+
+    def get_error_handler(self) -> ErrorHandler:
+        # Error status code mapping from Amplitude documentation: https://amplitude.com/docs/apis/analytics/export#status-codes
+        error_mapping = DEFAULT_ERROR_MAPPING | {
+            400: ErrorResolution(
+                response_action=ResponseAction.FAIL,
+                failure_type=FailureType.config_error,
+                error_message="The file size of the exported data is too large. Shorten the time ranges and try again. The limit size is 4GB. Provide a shorter 'request_time_range' interval.",
+            ),
+            403: ErrorResolution(
+                response_action=ResponseAction.FAIL,
+                failure_type=FailureType.config_error,
+                error_message="Access denied due to lack of permission or invalid API/Secret key or wrong data region.",
+            ),
+            404: ErrorResolution(
+                response_action=ResponseAction.IGNORE,
+                failure_type=FailureType.config_error,
+                error_message="No data available for the time range requested.",
+            ),
+            504: ErrorResolution(
+                response_action=ResponseAction.FAIL,
+                failure_type=FailureType.config_error,
+                error_message="The amount of data is large and may be causing a timeout. For large amounts of data, the Amazon S3 destination is recommended. Refer to Amplitude documentation for information on setting up the S3 destination.",
```

Comment:
nit: did you find a direct link to the documentation for setting up S3 for amplitude? if so lets add it somewhere here

Reply:
Added